src/Microsoft.AspNetCore.Authentication.OpenIdConnect/Events/IOpenIdConnectEvents.cs

```diff
@@ -38,17 +28,22 @@ public interface IOpenIdConnectEvents : IRemoteAuthenticationEvents
         /// <summary>
         /// Invoked before redirecting to the identity provider to authenticate.
         /// </summary>
-        Task RedirectToAuthenticationEndpoint(RedirectContext context);
+        Task RedirectToIdentityProvider(RedirectContext context);
```

Comment:
@PinpointTownes Here there's some more background. Although people who are familiar with the protocol details should be able to consume the object model, familiarity and full understanding of the protocol stack should not be a strict requirement. Mentioning "authorization" would be misleading given that that's not the main reason for which someone would use the middleware (signing in in itself does not entail authorization) and only people who know about the authorization endpoint and its role in those grants would understand what this is about. "AuthenticationEndpoint" would be also confusing, given that authentication is not the only reason for which this roundtrip is made (consent alone is a possibility) and the assonance with "authorization" would be a source of further confusion. Sticking with the old name has the advantage of describing a phase of the exchange without tying to the details  of this specific protocol - a developer who needs to augment the message back to the authority will have a coherent place to work in across all redirect based protocols.
@Tratcher re:other middlewares, that's the usual discussion. The other OAuth2 based,. provider specific middlewares are actually ALWAYS redirecting for authorization purposes - you are piggybacking on that for "simulating" sign in but the semantic of the operation is NEVER sing in (sign in with the provider happens as a side effect and it is performed transparently from your middleware, your entry and exit points are of an authorization request regardless of what happens in the middle).